framework/src/dirackernels/DiracKernel.C

```diff
@@ -38,6 +38,14 @@ DiracKernel::validParams()
       "has been added before. If this option is set to false duplicate points are retained"
       "and contribute to residual and Jacobian.");
 
+  MooseEnum point_not_found_behavior("ERROR WARNING IGNORE", "IGNORE");
+  params.addParam<MooseEnum>(
+      "point_not_found_behavior",
+      point_not_found_behavior,
+      "By default (IGNORE), it is ignored if an added point cannot be located in the "
+      "specified subdomains. If this option is set to ERROR, this situation will result in an "
+      "error. If this option is set to WARNING, then a warning will be issued.");
```

Comment:
Did you decide to go with `IGNORE` for backwards compatibility? If so then I don't think we've implemented this block-restrictability correctly. I know @friedmud 's vote was for error by default. I think that would be my vote as well. As a potential user I would want to know if my point didn't fall in my subdomain, as opposed to my simulation silently chugging along.

Reply:
Marking as resolved because Alex changed his mind -- see below.